addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -82,17 +82,26 @@ function _buildTitleElement(title) {
  * @param {string} src
  * @returns {HTMLElement}
  */
+const _buildImgElementCache = {};
 async function _buildImgElement(src) {
-    if (src.split('.').pop() === 'svg') {
-        const response = await window.fetch(src);
-        const text = await response.text();
-        const parser = new window.DOMParser();
-        const xmlDoc = parser.parseFromString(text, 'text/xml');
-        return xmlDoc.getElementsByTagName('svg')[0];
+
+    if (!(src in _buildImgElementCache)) {
+        _buildImgElementCache[src] = (async () => {
+            if (src.split('.').pop() === 'svg') {
+                const response = await window.fetch(src);
+                const text = await response.text();
+                const parser = new window.DOMParser();
+                const xmlDoc = parser.parseFromString(text, 'text/xml');
+                return xmlDoc.getElementsByTagName('svg')[0];
+            } else {
+                const imgEl = document.createElement('img');
+                imgEl.src = src;
+                return imgEl;
+            }
```

Comment:
You are still not duplicating the elements, so it can't work:
- Drop text snippet -> click on it -> see the "Content width" option icons
- Repeat the operation with another text snippet
- Now reclick on the first snippet -> the "Content width" option icons are gone (because they have been used for the other snippet options)

Reply:
Odoo JS is really a dark forest I cannot walk alone :smiley: 

Fixed:
```
    const node = await _buildImgElementCache[src];
    return node.cloneNode(true);
```

Reply:
To work on this pull request I also read a tutorial to Promise/async/await, because I mostly worked in es5